core/modules/replica/ConfigurationSchema.h

```diff
@@ -0,0 +1,206 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_CONFIGURATIONSCHEMA_H
+#define LSST_QSERV_REPLICA_CONFIGURATIONSCHEMA_H
+
+// System headers
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+
+// Forward declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+    class Configuration;
+namespace database {
+namespace mysql {
+    class Connection;
+}}  // namespace database::mysql
+}}} // namespace lsst::qserv::replica
+
+// This header declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * This utility class ConfigurationSchema encaposulates the JSON and MySQL schemas
+ * and operations over the schemas of the Configuration service. Both schemas are
+ * put into this class to facilitate schema changes tracking, unit and integration
+ * testing, as well as other actions that may involve the schemas, such as schema
+ * evolution, schema verification, etc.
+ *
+ * @note This class is the primary source of the schemas for the given version.
+ */
+class ConfigurationSchema {
+public:
+    /// The current version number required by the application.
+    static int const version;
+
+    /// @return A documentation string for the specified parameter or the empty string
+    ///   if none is available in the schema.
+    static std::string description(std::string const& category, std::string const& param);
+
+    /// @return A 'true' if the parameter can't be modified via the 'set' methods
+    ///   of the Configuration class. This information is used by class Configuration
+    ///   to validate the parameters.
+    static bool readOnly(std::string const& category, std::string const& param);
+
+    /// @return A 'true' if the parameter represents the security context (passwords,
+    ///   authorization keys, etc.). Parameters possesing this attribute are supposed
+    ///   to be used with care by the dependent automation tools to avoid exposing
+    ///   sensitive information in log files, reports, etc.
+    static bool securityContext(std::string const& category, std::string const& param);
```

Comment:
This is impossible in the current implementation because `schema::securityContext("databsae", "password")` will throw an exception for non-existing parameter. So, it's guaranteed to work...at least in your example. Please, see the implementation of the method `schema::securityContext`.

Reply:
Though, I can see people doing crazy things that would break any API. There is a class of folks known as "hackers" for whom it's an obsession or a profession to break things that are supposed to be robust. I don't think we're developing the software that would be of any interest to those. In our case, the biggest thread is our own istakes.